src/main/java/cs2103/v15_1j/jimjim/searcher/KeywordFilter.java

```diff
@@ -20,19 +20,21 @@ public KeywordFilter(List<String> keywords) {
     @Override
     public boolean check(TaskEvent taskEvent) {
     	List<String> wordsInName = Arrays.asList(taskEvent.getName().split(" "));
-    	for (String keyword : keywords) {
-    		boolean found = false;
-    		for (String word : wordsInName) {
-    			if (word.equalsIgnoreCase(keyword) || 
-    				Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
-    				found = true;
-    				break;
+    	boolean first = true;
+    	for (String word : wordsInName) {
+    		for (String keyword : keywords) {
+    			if (first) {
+        			if (Character.toLowerCase(word.charAt(0)) == Character.toLowerCase(keyword.charAt(0))) {
```

Comment:
This still does not fix the problem. The task `milk Shaun` will still show up when Jim searches for `meet`